src/main/java/net/starlark/java/eval/Printer.java

```diff
@@ -398,7 +400,12 @@ public static void formatWithList(Printer printer, String pattern, List<?> argum
           continue;
 
         default:
-          throw new IllegalStateException("unreachable");
+          // The call to Starlark.repr doesn't cause an infinite recursion
+          // because it's only used to format a string properly.
+          throw new MissingFormatWidthException(
+              String.format(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;+&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 406. Consider formatting this data with &#39;%d&#39; instead.